tools/tracegen/Port.hpp

```diff
@@ -56,6 +56,9 @@
 #define OS_ENCODING_WC_FLAGS 0
 #define UNICODE_BUFFER_SIZE EsMaxPath
 #else /* defined(OMR_OS_WINDOWS) */
+#if !defined(__cplusplus) || (__cplusplus < 201103L)
+#define snprintf(buf, buf_size, format, ...) sprintf(buf, format, __VA_ARGS__)
+#endif /* !defined(__cplusplus) || (__cplusplus < 201103L) */
```

Comment:
This file should only be included in C++ contexts so `!defined(__cplusplus)` should be redundant.
Same comment for this pattern in other C++ contexts.